src/openfermion/__init__.py

```diff
@@ -17,243 +17,15 @@
-    amplitude_damping_channel,
-    anticommutator,
-    bch_expand,
-    boson_ladder_sparse,
-    boson_operator_sparse,
-    chemist_ordered,
-    commutator,
-    count_qubits,
-    dephasing_channel,
-    depolarizing_channel,
-    double_commutator,
-    eigenspectrum,
-    erpa_eom_hamiltonian,
-    error_bound,
-    error_operator,
-    expectation,
-    expectation_computational_basis_state,
-    fourier_transform,
-    freeze_orbitals,
-    gaussian_state_preparation_circuit,
-    generate_linear_qubit_operator,
-    generate_parity_permutations,
-    geometry_from_pubchem,
-    get_chemist_two_body_coefficients,
-    get_density_matrix,
-    get_file_path,
-    get_gap,
-    get_ground_state,
-    get_linear_qubit_operator_diagonal,
-    group_into_tensor_product_basis_sets,
-    haar_random_vector,
-    hartree_fock_state_jellium,
-    hermitian_conjugated,
-    inline_sum,
-    inner_product,
-    inverse_fourier_transform,
-    is_hermitian,
-    is_identity,
-    jordan_wigner_sparse,
-    jw_configuration_state,
-    jw_get_gaussian_state,
-    jw_get_ground_state_at_particle_number,
-    jw_hartree_fock_state,
-    jw_number_restrict_operator,
-    jw_number_restrict_state,
-    jw_slater_determinant,
-    jw_sz_restrict_operator,
-    jw_sz_restrict_state,
-    kronecker_delta,
-    lambda_norm,
-    load_operator,
-    low_depth_second_order_trotter_error_bound,
-    low_depth_second_order_trotter_error_operator,
-    low_rank_two_body_decomposition,
-    majorana_operator,
-    make_reduced_hamiltonian,
-    map_one_hole_dm_to_one_pdm,
-    map_one_pdm_to_one_hole_dm,
-    map_particle_hole_dm_to_one_pdm,
-    map_particle_hole_dm_to_two_pdm,
-    map_two_hole_dm_to_one_hole_dm,
-    map_two_hole_dm_to_two_pdm,
-    map_two_pdm_to_one_pdm,
-    map_two_pdm_to_particle_hole_dm,
-    map_two_pdm_to_two_hole_dm,
-    module_importable,
-    normal_ordered,
-    number_operator,
-    pauli_exp_to_qasm,
-    prepare_one_body_squared_evolution,
-    preprocess_lcu_coefficients_for_reversible_sampling,
-    prune_unused_indices,
-    qubit_operator_sparse,
-    random_antisymmetric_matrix,
-    random_diagonal_coulomb_hamiltonian,
-    random_hermitian_matrix,
-    random_interaction_operator,
-    random_quadratic_hamiltonian,
-    random_qubit_operator,
-    random_unitary_matrix,
-    reduce_number_of_terms,
-    reorder,
-    s_minus_operator,
-    s_plus_operator,
-    s_squared_operator,
-    save_operator,
-    singlet_erpa,
-    slater_determinant_preparation_circuit,
-    sparse_eigenspectrum,
-    sx_operator,
-    sy_operator,
-    sz_operator,
-    taper_off_qubits,
-    trotter_operator_grouping,
-    trotterize_exp_qubop_to_qasm,
-    uccsd_convert_amplitude_format,
-    uccsd_generator,
-    uccsd_singlet_generator,
-    uccsd_singlet_get_packed_amplitudes,
-    uccsd_singlet_paramsize,
-    up_then_down,
-    variance,
-    wedge,
-)
-
-from ._version import \
-    __version__
+from openfermion.chem import *
```

Comment:
in what sense?  Like list all them out?  I think that is redundant. `*` was made for this purpose.  Especially since we explicitly list them in the inits.  If there is a docs reason to list them here then I'm fine with listing.  Outside of that I think it isn't necessary.

Reply:
By my understanding import * is typically considered bad practice in python - https://stackoverflow.com/questions/2386714/why-is-import-bad . I also find it quicker when I'm in an editor that has openfermion to pop open __init__.py and ctrl-f a specific function to find its location than to look in the docs.

But that being said I'm not particularly tied to this. Also happy to add the text myself if you want.

Reply:
I certainly agree with the examples in the stackoverflow post.  Being explicit when you use the functionality in a module or in a routine is always a good idea.  But when lifting imports from __init__.py from lower modules to the top-level-directory I don't see a problem.   My guiding principle here is that we should explicitly import when it obviously helps the programmer or reader of the code.  Lifting module namespaces with `import *` actually seems like a good  shorthand for the code reader.  We are truly importing everything from the submodule __init__.    Let's keep this as is for now.  If python-norms change we can update later.